index.bs

```diff
@@ -13244,6 +13264,162 @@ the exact steps to take if <dfn>an exception was thrown</dfn>, or by explicitly
 </div>
 
 
+<h3 id="synthetic-module-records">Synthetic module records</h3>
+
+A <dfn export>Synthetic Module Record</dfn> is used to represent information about a module that is
+defined by specifications.
+Its exports are derived from a pair of lists, of string keys and of ECMAScript values.
+The set of exported names is static, and determined at creation time (as an argument to
+[$CreateSyntheticModule$]), while the set of exported values can be changed over time using
+[$SetSyntheticModuleExport$].
+It has no imports or dependencies.
+
+Note: A [=Synthetic Module Record=] could be used for defining a variety of module types: for
+example, built-in modules, or JSON modules, or CSS modules.
+
+In addition to the [=Module Record Fields=], Synthetic Module Records have the additional fields
+listed below.
+Each of these fields is initially set in [$CreateSyntheticModule$].
+
+<table id="table-synthetic-module-record-fields" class="data">
+<caption>Additional Fields of Synthetic Module Records</caption>
+<thead>
+  <tr>
+    <th>Field Name
+    <th>Value Type
+    <th>Meaning
+<tbody>
+  <tr>
+    <td>\[[ExportNames]]
+    <td>List of String
+    <td>A List of all names that are exported.
+  <tr>
+    <td>\[[EvaluationSteps]]
+    <td>An abstract operation
+    <td>An abstract operation that will be performed upon evaluation of the module, taking the
+        [=Synthetic Module Record=] as its sole argument.
+        These will usually set up the exported values, by using [$SetSyntheticModuleExport$].
+        They must not modify \[[ExportNames]].
+        They may return an abrupt completion.
+</table>
+
+<h4>CreateSyntheticModule</h4>
+
+<div algorithm>
+The abstract operation <dfn abstract-op>CreateSyntheticModule</dfn>(|exportNames|, |evaluationSteps|, |realm|, |hostDefined|) creates a [=Synthetic Module Record=] based upon
+the given exported names and evaluation steps.
+It performs the following steps:
+
+    1.  Return [=Synthetic Module Record=] {
+            \[[Realm]]: |realm|,
+            \[[Environment]]: <emu-val>undefined</emu-val>,
+            \[[Namespace]]: <emu-val>undefined</emu-val>,
+            \[[HostDefined]]: |hostDefined|,
+            \[[ExportNames]]: |exportNames|,
+            \[[EvaluationSteps]]: |evaluationSteps|
+        }.
+</div>
+
+Note: It seems we could set up the environment either here or in Instantiate().
```

Comment:
Maybe rephrase a bit; the current note is written as more of a note to the reviewer, not to the reader. I'd do

> Note: We could set up the \[[Environment]] either here or in [=Instantiate()=]. It is done in Instantiate() for symmetry with [=Source Text Module Records=], but there is no observable difference.